src/dpki.rs

```diff
@@ -0,0 +1,377 @@
+// finally the random salt.5
+pub struct EncryptedSeed(pub [u8; ENCRYPTED_SEED_SIZE]);
+
+/*
+ * from holochain_dpki::seed
+ *
+pub(crate) fn pw_enc_zero_nonce(
+    data: &[u8],
+    passphrase: &mut SecBuf,
+    config: Option<PwHashConfig>,
+) -> HcResult<EncryptedData> {
+    let mut salt = SecBuf::with_insecure(SALTBYTES);
+    salt.randomize();
+    let mut nonce = SecBuf::with_insecure(NONCEBYTES);
+    nonce.write(0, &[0; NONCEBYTES])?;
+    let data = pw_enc_base(data, passphrase, &mut salt, &mut nonce, config)?;
+    Ok(data)
+}
+
+/// Private general wrapper of pw_enc
+fn pw_enc_base(
+    data: &mut SecBuf,
+    passphrase: &mut SecBuf,
+    mut salt: &mut SecBuf,
+    mut nonce: &mut SecBuf,
+    config: Option<PwHashConfig>,
+) -> HcResult<EncryptedData> {
+    let mut secret = SecBuf::with_secure(kx::SESSIONKEYBYTES);
+    let mut cipher = SecBuf::with_insecure(data.len() + aead::ABYTES);
+    pw_hash(passphrase, &mut salt, &mut secret, config)?;
+    aead::enc(data, &mut secret, None, &mut nonce, &mut cipher)?;
+
+    let salt = salt.read_lock().to_vec();
+    let nonce = nonce.read_lock().to_vec();
+    let cipher = cipher.read_lock().to_vec();
+    // Done
+    Ok(EncryptedData {
+        salt,
+        nonce,
+        cipher,
+    })
+}
+ */
+
+pub trait SeedData
+where
+    Self: Sized,
+{
+    fn from_bytes(b: &[u8]) -> Result<Self, Error>;
+    fn as_bytes<'a>(&'a self) -> &'a [u8];
+}
+
+impl SeedData for Seed {
+    fn from_bytes(b: &[u8]) -> Result<Self, Error> {
+        ensure!(b.len() == SEED_SIZE, "Incorrect length for Seed");
+        let mut seed_bytes = [0u8; SEED_SIZE];
+        seed_bytes.copy_from_slice(&b[..SEED_SIZE]);        
+        Ok(Self(seed_bytes))
+    }
+    fn as_bytes<'a>(&'a self) -> &'a [u8] {
+        &self.0
+    }
+}
+
+impl Seed {
+    pub fn encrypt(
+        &self,
+        _password: &str, // TODO: implmenent aead encryption!
+    ) -> Result<EncryptedSeed, Error> {
+        /*
+        let encrypted_data =
+            pw_enc_zero_nonce(&mut self.seed_mut().buf, &mut passphrase_buf, config)?;
+        Ok(EncryptedSeed::new(encrypted_data, self.seed().kind.clone()))
+         */
+        let mut enc_seed = [0u8; ENCRYPTED_SEED_SIZE];
+        enc_seed[..SEED_SIZE].copy_from_slice(self.as_bytes());
+        EncryptedSeed::from_bytes(&enc_seed)
+    }
+}
+
+impl SeedData for EncryptedSeed {
+    fn from_bytes(b: &[u8]) -> Result<Self, Error> {
+        ensure!(b.len() == ENCRYPTED_SEED_SIZE, "Incorrect length for EncryptedSeed");
+        let mut enc_seed_bytes = [0u8; ENCRYPTED_SEED_SIZE];
+        enc_seed_bytes.copy_from_slice(&b[..ENCRYPTED_SEED_SIZE]);        
+        Ok(Self(enc_seed_bytes))
+    }
+    fn as_bytes<'a>(&'a self) -> &'a [u8] {
+        &self.0
+    }
+}
+
+
+/// Create seeds from, or render to Mnemonics, for seeds that are multiples of SEED_SIZE.
+pub trait MnemonicableSeed
```

Comment:
Not sure why this is a trait.

Reply:
Mostly for compatibility with the holochain-rust `dpki` APIs, when we start using them.